### ISSUE TITLE
Twisted's README and other top-level documents should be in rst format

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8054]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)| @hawkowl reported|
|-|-|
|Trac ID|trac#8054|
|Type|enhancement|
|Created|2015-10-20 01:54:27Z|
|Branch|https://github.com/twisted/twisted/tree/rst-toplevel-8054|

Then we can have BADGES. We can have MARKUP. We can have AWESOMENESS. Also, it makes us not look like we're from the 80s on GitHub.

<details><summary>Searchable metadata</summary>

```
trac-id__8054 8054
type__enhancement enhancement
reporter__hawkowl hawkowl
priority__normal normal
milestone__None None
branch__branches_rst_toplevel_8054 branches/rst-toplevel-8054
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1445306067321411 1445306067321411
changetime__1447587892984422 1447587892984422
version__None None
owner__hawkowl hawkowl

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_3"></a>|@hawkowl commented|
|-|-|

I like this. Please review. Builds spun.

**Comment by twisted-trac** [comment]
|[<img alt="adiroiban's avatar" src="https://avatars.githubusercontent.com/u/204609?s=50" width="50" height="50">](https://github.com/adiroiban)<a name="note_4"></a>|@adiroiban set owner to @hawkowl|
|-|-|

So.. the nice looking README is here

https://github.com/twisted/twisted/tree/rst-toplevel-8054

and then for INSTALL

https://github.com/twisted/twisted/blob/rst-toplevel-8054/INSTALL.rst


It looks like this is not only a conversion to RST, but also a refactor/update of the content... It would have been much easier to review if the RST conversion would have been done in a commit and then content changes in another.

-----------

For INSTALL.rst

1. To me, it looks like Arch package name is the same as Debian/Ubuntu. Do we need a separate list item for Arch ?... should we include the name for Fedora/CentOS/RHEL ?

All good.. happy to see that we have wheels for Windows :)

---------

For README.rst

1. Do we need / want to duplicate all the install part in README? I think that ```pip install twisted``` followed by a link to INSTALL.rst should be enought.
1. For unit test, I think that we should still talk first about ./bin/trial ... but trial is only important for Twisted dev and this is just the readme... which don't talk about  #twisted-dev 
1. For tox, test will fail from many, many reasons ... like lack of python version, or lack of headers for that python version... lack of an installed compiler... pyflakes/apidocs/twistedchecker alwasy being broken :( 
1. For the unit test section I think that we should link to a page from our dev docs
1. Maybe the final LICENSE reference should be converted to a link.

------

Is is ok not to update twisted/python/_release.py with the new name for the README file ?

I was expecting for the release test to fail... maybe they should be updated.


----

Please check my comments and merge.

Thanks!

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_5"></a>|@hawkowl commented|
|-|-|

Thanks Adi. Fixed up some of those issues.

```bin/trial``` is a developer tool, it shouldn't be used by people that just want to verify Twisted works. Tox is a much better tool for that.

Will merge.

**Comment by twisted-trac** [comment]
|[<img alt="hawkowl's avatar" src="https://avatars.githubusercontent.com/u/3307100?s=50" width="50" height="50">](https://github.com/hawkowl)<a name="note_6"></a>|@hawkowl set status to `closed`|
|-|-|

(In [46217]) Merge rst-toplevel-8054: Change the top level documents to be Restructured Text, and make them otherwise nicer

Author: hawkowl
Reviewer: adiroiban
Fixes: [#8054](https://github.com/twisted/twisted/issues/8054)
